src/GafferOSL/OSLObject.cpp

```diff
@@ -290,20 +295,27 @@ void OSLObject::compute( Gaffer::ValuePlug *output, const Gaffer::Context *conte
 			static_cast<StringPlug *>( output )->setToDefault();
 			return;
 		}
+		std::string primitiveVariablesToResample;
+
+		const OSLShader *shader = runTimeCast<const OSLShader>( shaderPlug()->source()->node() );
+		ConstShadingEnginePtr shadingEngine = shader ? shader->shadingEngine() : nullptr;
```

Comment:
The code below seems to assume that the `nullptr` case will never actually occur. Can we guarantee that?